.github/workflows/generate_sdk_ref.yml

```diff
@@ -0,0 +1,106 @@
+name: Generate SDK references
+
+on:
+  workflow_dispatch:
+  workflow_call:
+  # Remove after testing
+  push:
+    branches:
+      - generate-api-reference-for-code-interpreter-sdk-e2b-1235
+
+concurrency: ${{ github.workflow }}-${{ github.ref }}
+
+permissions:
+  contents: write
+
+jobs:
+  is_new_sdk_ref:
+    name: Is new SDK reference?
+    runs-on: ubuntu-latest
+    outputs:
+      new_sdk_ref: ${{ steps.sdk-changes.outputs.new_sdk_ref }}
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+      
+      - name: Check if SDK changes
+        id: sdk-changes
+        run: |
+          IS_NEW_sdk_REF=$(./.github/scripts/is_new_sdk_ref.sh)
+          echo "new_sdk_ref=$IS_NEW_sdk_REF" >> "$GITHUB_OUTPUT"
+
+  sdk-changes:
+    name: SDK changes
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Install pnpm
+        uses: pnpm/action-setup@v3
+        id: pnpm-install
+        with:
+          version: 9.5
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: "20.x"
+          registry-url: "https://registry.npmjs.org"
+          cache: pnpm
+          cache-dependency-path: pnpm-lock.yaml
+
+      - name: Configure pnpm
+        run: |
+          pnpm config set auto-install-peers true
+          pnpm config set exclude-links-from-lockfile true
+
+      - name: Install dependencies
+        run: pnpm install --frozen-lockfile
+
+      - name: Set up Python
+        id: setup-python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.8"
+
+      - name: Install and configure Poetry
+        uses: snok/install-poetry@v1
+        with:
+          version: 1.5.1
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+          installer-parallel: true
+
+      - name: Install dependencies
+        working-directory: ./python
+        run: poetry install --no-interaction --no-root
+
+      - name: Generate Python SDK reference
+        id: python-sdk-ref
+        working-directory: ./python
+        run: |
+          source .venv/bin/activate
+          ./scripts/generate_sdk_ref.sh
+
+      - name: Generate JS SDK reference
+        id: js-sdk-ref
+        working-directory: ./js
+        run: ./scripts/generate_sdk_ref.sh
```

Comment:
I would move those in release workflow.

Also I would probably add it as a script to `package.json` and then call 
```
pnpm run --recursive generate-ref
```
or something like that ([recursive](https://pnpm.io/cli/recursive))

Similarly as you already have in [here](https://github.com/e2b-dev/E2B/blob/main/packages/python-sdk/package.json#L11).